salt/orchestrate/aws/map_templates/edx.yml

```diff
@@ -9,9 +9,17 @@ edx:
           SubnetId: {{ subnetids[id_num % num_subnets] }}
           SecurityGroupId: {{ securitygroupid }}
       tag:
+        business_unit: {{ business_unit }}
         environment: {{ environment_name }}
+        purpose: {{ purpose }}
       grains:
+        business_unit: {{ business_unit }}
         environment: {{ environment_name }}
+        {% if app_type not in purpose %}
+        {% set purpose = purpose[:-len(purpose.split('-')[-1])] +app_type %}
+        purpose:
+          - {{ purpose }}
```

Comment:
This should be a string rather than a list. I don't think we have a use case for needing more than one purpose on a given host.